src/firefly/js/ui/FireflyRoot.jsx

```diff
@@ -1,31 +1,68 @@
-import {CssBaseline, ScopedCssBaseline, CssVarsProvider, useColorScheme, GlobalStyles} from '@mui/joy';
-import {getTheme} from './ThemeSetup.js';
-
-
+import React, {useEffect} from 'react';
+import {ScopedCssBaseline, extendTheme, CssVarsProvider, useColorScheme, GlobalStyles} from '@mui/joy';
 
+import {defaultTheme} from './ThemeSetup.js';
+import {getAppOptions} from '../core/AppDataCntlr.js';
+import {logger} from '../util/Logger.js';
 
+import '@fontsource/inter/200.css'; // Lighter
+import '@fontsource/inter/300.css'; // Light
+import '@fontsource/inter/400.css'; // Regular
+import '@fontsource/inter/500.css'; // Medium
+import '@fontsource/inter/600.css'; // Semi-Bold
+import '@fontsource/inter/700.css'; // Bold
+import '@fontsource/inter/800.css'; // Bolder
 
 export function FireflyRoot({children}) {
-    const newTheme= getTheme();
-    const useBaseline= false;
+
+    const customTheme = getAppOptions().theme?.customized?.();
+    const theme = extendTheme(customTheme || defaultTheme());
 
     return (
-        <CssVarsProvider theme={newTheme}>
-            {useBaseline && <CssBaseline/>}
+        <CssVarsProvider defaultMode='system' theme={theme}>
             <GlobalStyles styles={{
-                    // CSS object styles
-                    html: {fontSize:'87.5%'}
-                }}/>
-            <App>{children}</App>
+                html: {fontSize:'87.5%'}
+            }}/>
+            <ScopedCssBaseline>
+                <App>{children}</App>
+            </ScopedCssBaseline>
         </CssVarsProvider>
     );
 }
 
 function App({children}) { // provide a way to experiment with light and dark theme
-    const { mode, setMode } = useColorScheme();
-   // setMode('dark');
-    setMode('light');
+
+    // Firefly is setup to take the device's color mode
+    // To set device mode on your mac; AppleMenu -> System Settings -> Appearance
+
+    const { activeMode, setMode } = useColorMode();
+
+    const colorMode = getAppOptions().theme?.colorMode?.();
+    // if colorMode is set; must use this mode
+    // To add mode switching; make sure colorMode is not set.
+    // Then, add a component(Button) to trigger 'setMode'.
+    // setMode store the value in localStorage.  It will use this mode
+    // until that value is cleared or kill/restart browser.
+
+    logger.info({activeMode, colorMode});
+
+    useEffect(() => {
+        // only setMode when colorMode is different from the current activeMode
+        if (colorMode && colorMode !== activeMode) {
+            setMode(colorMode);
+        }
+    }, [colorMode, activeMode]);
     return ( children );
 }
 
+
+/**
+ * A wrapper around useColorScheme to return the current active mode, which should be either light or dark
+ * @return {object}
+ */
+export function useColorMode() {
+    const { mode, systemMode, ...rest } = useColorScheme();
+    const activeMode = mode === 'system' ? systemMode : mode;
+    return {activeMode, mode, systemMode, ...rest};
```

Comment:
I feel like this is incomplete. I don't think system mode works (I tested with my mac). Even it is did it does not have the event listeners to change it when system mode is set to auto or changes.

some googling show we can do like this.
```
 window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', () => /*something*/)
```
I think it is worth looking into `window.matchMedia` then `addEventListener`

Reply:
This works fine for me.  I can demonstrate it for you.  The culprit is `setMode` used in existing code(fixed in my PR).  I tried to explained this in the comments.  `setMode` uses browser local storage.  Once set, it will ALWAYS take that value regardless of what your device setting is.

This is the intended design for our color mode.
```
    const colorMode = getAppOptions().theme?.colorMode?.();
    // if colorMode is set; must use this mode
    // To add mode switching; make sure colorMode is not set.
    // Then, add a component(Button) to trigger 'setMode'.
    // setMode store the value in localStorage.  It will use this mode
    // until that value is cleared or kill/restart browser.
```